modules/nobidBidAdapter.js

```diff
@@ -0,0 +1,290 @@
+nobid.bidResponses = nobid.bidResponses || {};
+nobid.renderTag = function(doc, id, win) {
+  log('nobid.renderTag()', id);
+  var bid = nobid.bidResponses['' + id];
+  if (bid && bid.adm2) {
+    log('nobid.renderTag() found tag', id);
+    var markup = bid.adm2;
+    doc.write(markup);
+    doc.close();
+    return;
+  }
+  log('nobid.renderTag() tag NOT FOUND *ERROR*', id);
+}
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+  /**
+ * Determines whether or not the given bid request is valid.
+ *
+ * @param {BidRequest} bid The bid params to validate.
+ * @return boolean True if this is a valid bid, and false otherwise.
+ */
+  isBidRequestValid: function(bid) {
+    log('isBidRequestValid', bid);
+    return !!bid.params.siteId;
+  },
+  /**
+ * Make a server request from the list of BidRequests.
+ *
+ * @param {validBidRequests[]} - an array of bids
+ * @return ServerRequest Info describing the request to the server.
+ */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    function resolveEndpoint() {
+      var ret = 'https://ads.servenobid.com/';
+      var env = (typeof utils.getParameterByName === 'function') && (utils.getParameterByName('nobid-env'));
+      if (!env) ret = 'https://ads.servenobid.com/';
+      else if (env == 'beta') ret = 'https://beta.servenobid.com/';
+      else if (env == 'dev') ret = '//localhost:8282/';
+      else if (env == 'qa') ret = 'https://qa-ads.nobid.com/';
+      return ret;
+    }
+    var buildEndpoint = function() {
+      return resolveEndpoint() + 'adreq?cb=' + Math.floor(Math.random() * 11000);
+    }
+    log('buildRequests', validBidRequests);
+    if (!validBidRequests || validBidRequests.length <= 0) {
+      log('Empty validBidRequests');
+      return;
+    }
+    const payload = nobidBuildRequests(validBidRequests, bidderRequest);
+    if (!payload) return;
+    const payloadString = JSON.stringify(payload).replace(/'|&|#/g, '')
+    const endpoint = buildEndpoint();
+    return {
+      method: 'POST',
+      url: endpoint,
+      data: payloadString,
+      bidderRequest
+    };
+  },
+  /**
+     * Unpack the response from the server into a list of bids.
+     *
+     * @param {ServerResponse} serverResponse A successful response from the server.
+     * @return {Bid[]} An array of bids which were nested inside the server.
+     */
+  interpretResponse: function(serverResponse, bidRequest) {
+    log('interpretResponse', serverResponse);
+    log('interpretResponse', bidRequest);
+    return nobidInterpretResponse(serverResponse.body, bidRequest);
+  },
+
+  /**
+     * Register the user sync pixels which should be dropped after the auction.
+     *
+     * @param {SyncOptions} syncOptions Which user syncs are allowed?
+     * @param {ServerResponse[]} serverResponses List of server's responses.
+     * @return {UserSync[]} The user syncs which should be dropped.
+     */
+  getUserSyncs: function(syncOptions, serverResponses) {
```

Comment:
FYI:

GDPR information can be passed into the getUserSyncs function and passed along to your sync endpoint.

See https://github.com/prebid/Prebid.js/blob/master/modules/rubiconBidAdapter.js#L602 as an example.

Maybe add it in a future PR if needed?

Reply:
This is done